maat-court-data-api/src/test/java/gov/uk/courtdata/eform/service/EformsHistoryServiceTest.java

```diff
@@ -0,0 +1,54 @@
+package gov.uk.courtdata.eform.service;
+
+import gov.uk.courtdata.eform.repository.EformsHistoryRepository;
+import gov.uk.courtdata.eform.repository.entity.EformsHistory;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.List;
+
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+public class EformsHistoryServiceTest {
+
+    @InjectMocks
+    private EformsHistoryService eformsHistoryService;
+    @Mock
+    private EformsHistoryRepository eformsHistoryRepository;
+
+    @Test
+    void shouldRetrieveEformDecisionHistoryForGivenUSN() {
+        EformsHistory eformsHistory = EformsHistory.builder().id(1).usn(123).action("Get").userCreated("test").build();
+        List<EformsHistory> eformsHistoryList = List.of(eformsHistory);
+        when(eformsHistoryRepository.findAllByUsn(anyInt())).thenReturn(eformsHistoryList);
+        eformsHistoryService.getAllEformsHistory(anyInt());
+        verify(eformsHistoryRepository, times(1)).findAllByUsn(anyInt());
+    }
+
+    @Test
+    void shouldRetrieveLatestEformDecisionHistoryForGivenUSN() {
+        EformsHistory eformsHistory = EformsHistory.builder().id(1).usn(123).action("Get").userCreated("test").build();
+        when(eformsHistoryRepository.findTopByUsnOrderByIdDesc(anyInt())).thenReturn(eformsHistory);
+        eformsHistoryService.getLatestEformsHistoryRecord(anyInt());
+        verify(eformsHistoryRepository, times(1)).findTopByUsnOrderByIdDesc(anyInt());
```

Comment:
I'm sure this is fine, just a general query for my understanding as I've not used anyInt before - If we provide the USN on line 35 here, shouldn't we use it in the following lines instead of anyInt()?

Reply:
updated with usn number